spring-batch-core/src/main/java/org/springframework/batch/core/step/tasklet/CallableTaskletAdapter.java

```diff
@@ -35,6 +35,21 @@
 
 	private Callable<RepeatStatus> callable;
 
+	/**
+	 * Create a new {@link CallableTaskletAdapter} instance.
+	 */
+	public CallableTaskletAdapter() {
+	}
+
+	/**
+	 * Create a new {@link CallableTaskletAdapter} instance.
+	 * @param callable the {@link Callable} to use
+	 */
+	public CallableTaskletAdapter(Callable<RepeatStatus> callable) {
+		setCallable(callable);
+		afterPropertiesSet();
```

Comment:
I referenced  the constructor of `JdbcTemplate`, which also provides both setter and constructor injection.

https://github.com/spring-projects/spring-framework/blob/master/spring-jdbc/src/main/java/org/springframework/jdbc/core/JdbcTemplate.java#L172